palette.js

```diff
@@ -2,86 +2,116 @@ class View {
   constructor(body) {
     this.body = body;
   }
-  onSubmit(cb) {
+  onKeyPress(cb) {
     this.body.onkeypress = cb;
   }
   onLoad(cb) {
     this.body.onload = cb;
-    console.log("loaded");
   }
 }
 
-class PaletteObject {
-  constructor(input, container, lockIcon, copyIcon, arrowIcon) {
+class PaletteView {
+  constructor(input, form, container, lockIcon, copyIcon, arrowIcon) {
     this.input = input;
+    this.form = form;
     this.container = container;
     this.lockIcon = lockIcon;
     this.copyIcon = copyIcon;
     this.arrowIcon = arrowIcon;
+    this.setupFormAction();
   }
-}
 
+  setupFormAction() {
+    this.form.onsubmit = (ev) => {
+      ev.preventDefault();
+      this.container.style.backgroundColor = this.input.value;
+    };
+  }
+  //Pass Color in to update field and background
+  static setColor() {
+    let hex = new Color(randomHex());
+    //updating field and background don't work here - I think it hasn't actually been created yet, because it's creating an array?
+    //left controler to make the changes, but it uses this method now.
+    return hex;
+  }
+}
 class PaletteController {
-  constructor(paletteObjectArr, submission) {
+  constructor(paletteObjectArr, keyPress) {
     this.paletteObjectArr = paletteObjectArr;
-    this.submission = submission;
-    this.submission.onSubmit(this.submit.bind(this));
-    this.submission.onLoad(this.load.bind(this));
+    this.keyPress = keyPress;
+    this.keyPress.onKeyPress(this.paletteShuffle.bind(this));
+    this.keyPress.onLoad(this.load.bind(this));
   }
-
-  submit(event) {
+  paletteShuffle(event) {
     if (event.keyCode == 32) {
       for (let i in this.paletteObjectArr) {
-        let hex = Color.randomColor().colorCode;
         const arrItem = this.paletteObjectArr[i];
-        arrItem.input.value = hex;
-        arrItem.container.style.backgroundColor = hex;
+        arrItem.input.value = PaletteView.setColor().colorCode;
+        arrItem.container.style.backgroundColor = PaletteView.setColor().colorCode;
```

Comment:
Got it. I think I was making a new instance of PaletteView.setColor instead of calling it on the arrItems, which is also a scope understanding issue. Scope trips me up a lot, but this video helps! I think I have it written to ticket spec now! Thanks!